models/User.go

```diff
@@ -1677,15 +1793,32 @@ func PerformOrderFillTransaction(ask *Ask, bid *Bid, stockTradePrice uint64, sto
 		bid.IsClosed = oldBidIsClosed
 
 		tx.Rollback()
-		return AskUndone, BidUndone, nil
+	}
+
+	if cashLeft < MINIMUM_CASH_LIMIT {
+		l.Errorf("Check1: Failed. Not enough cash. Returning reserved cash back to user.")
+		if err := bid.Close(tx); err != nil {
+			errorHelper("Unable to close bid. Rolling back. Error: %+v", err)
+			return AskUndone, BidUndone, nil
+		}
+		SaveBidCancelOrderTransaction(bid, biddingUser, tx)
+		if err := tx.Commit(); err != nil {
+			errorHelper("Error while commiting. Rolling back. Error: %+v", err)
+			return AskUndone, BidUndone, nil
+		}
+		go updateDataStreams(nil, nil, nil, nil)
```

Comment:
...

Reply:
This line has always been there, and I genuinely don't know why

Reply:
Just didn't wanna break anything..

https://github.com/delta/dalal-street-server/pull/252/files/470a3fff0d480adfb2390bf1a65a3ea20284b3d9#diff-a7cc9dd6c958d3bdf2d808ee43ab5e1dL1598

Reply:
ah i think that sends the "order is closed" notifs. looks super weird tho.

Reply:
Same. Thought it was best to leave it untouched lol